adrs/012-if-then-else.md

```diff
@@ -0,0 +1,99 @@
+# ADR 012 - If, Then and Else
+### Status: Draft
+### Josh Holbrook
+
+## Context
+
+It's time to implement conditional branching! But there's specification work
+to be done. We don't want to naively implement MSX BASIC's semantics for
+if/then/else, because it's really limited. However, we do want to retain some
+of the syntax.
+
+MSX BASIC supports the following forms:
+
+- `IF <condition> THEN <lineNo | instructions>`
+- `IF <condition> THEN <lineNo | instructions> ELSE <lineNo | instructions>`
+- `IF <condition> GOTO <lineNo>`
+- `IF <condition> GOTO <lineNo> ELSE <lineNo | instructions>`
+
+When supplied a lineNo, BASIC will jump to that line. Otherwise, it will run
+the listed instructions. Note that `Instructions` is what MSX BASIC calls the
+equivalents to Matanuska's "commands", and are likewise colon-separated.
+
+Also notable is that MSX BASIC does *not* support multi-line if statements.
+For example, the following is unsupported:
+
+```
+IF <condition> THEN
+  ...
+ELSE
+  ...
+ENDIF
+```
+
+There is, of course, no token equivalent to "end if" - we would have to decide
+on one. Sadly, the obvious choice - "end" - is currently meant to end execution
+of the program (ie, a program-level return). That decision isn't set in stone,
+though - alternatives include using `return` as in Lua (in fact, the command
```

Comment:
* `endif`/`end-if`/`end if` are valid choices worth considering
* can also use `fi` as in bash